src/desktop/apps/artwork/components/commercial/view.coffee

```diff
@@ -39,14 +40,23 @@ module.exports = class ArtworkCommercialView extends Backbone.View
   acquire: (e) ->
     e.preventDefault()
 
-    order = new PendingOrder
-    @form = new Form $form: @$('form'), model: order
+    loggedInUser = CurrentUser.orNull()
+    if loggedInUser?.hasLabFeature('New Buy Now Flow')
+      createOrder
```

Comment:
This feels like a 'side-effecty' way of invoking `createOrder`. It makes an async call to MP, and then promise is resolved there with a `location.assign` redirect.

Perhaps `createOrder` should be an async function, and in this file you do `cosnt order = await createOrder....; location.assign(...`. That might make it a bit clearer.